src/components/ProfilePage/Leaderboard/Leaderboard.tsx

```diff
@@ -45,14 +53,16 @@ export const Leaderboard = ({ address }: { address?: string }) => {
   const { data: leaderboardUserData }: { data: LeaderboardUserData } =
     useLeaderboardUser(address);
 
-  const leaderboardListLength =
-    meta?.pagination?.pagesLength || LEADERBOARD_LENGTH;
-
+  // set leaderboard list length to the number of pages only once
   useEffect(() => {
-    setCurrentPage((current) =>
-      Math.min(Math.max(current, 1), leaderboardListLength),
-    );
-  }, [leaderboardListLength]);
+    if (
+      meta?.pagination?.pagesLength &&
+      leaderboardListLength === LEADERBOARD_LENGTH
+    ) {
+      setLeaderboardListLength(meta.pagination.pagesLength);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Would prefer to solve the eslint instead disabling it

Reply:
of course, but i don't want `leaderboardListLength` to be a dependency because it changes how the app behaves, if you have an idea how to keep the same flow with making eslint happy please advise

Reply:
In my opinion, as this is what we just been talking about in Discord: There may be cases where dependency arrays not satisfy the linter. Fine to disable the linter, but add a short comment above.